File: code/jvm/main/kotlin/com/isel/leic/ps/ion_classcode/domain/Feedback.kt
package com.isel.leic.ps.ion_classcode.domain

/**
 * Feedback Domain Interface
 */
data class Feedback(
    val id: Int,
    val description: String,
    val label: String,
    val teamId: Int,
) {
    init {
        require(id > 0) { "Feedback id must be greater than 0" }
        require(description.isNotBlank()) { "Feedback description cannot be blank" }
        require(label.isNotBlank()) { "Feedback label cannot be blank" }
        require(teamId > 0) { "Team id must be greater than 0" }
    }
}

File: code/js/src/domain/Feedback.ts

export interface FeedbackDomain {
    id: number,
    description: string,
    label: string,
    teamId: number,
}

export class FeedbackDomain {
    constructor(
        id: number,
        description: string,
        label: string,
        teamId: number,
    ) {
        this.id = id
        this.description = description
        this.label = label
        this.teamId = teamId
    }
}